### PR TITLE
Add docstrings to classes and methods in expression.py

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: python/hail/expr/expression.py [hunk 1]
```diff
@@ -305,74 +307,230 @@ def __ne__(self, other):
...
+
+        Returns
+        -------
+        :py:class:`CollectionExpression`
+            Expression of the same type as the callee.
+        """
         return self._bin_lambda_method("filter", f, self._type.element_type, lambda t: self._type)
 
     def find(self, f):
+        """Returns a the first element where `f` returns `True`.
```
Comment (patrick-schultz) [inline python/hail/expr/expression.py]: Returns the first ...

File: python/hail/expr/expression.py [hunk 2]
```diff
@@ -305,74 +307,230 @@ def __ne__(self, other):
...
+        """
         return self._bin_lambda_method("find", f, self._type.element_type, lambda t: self._type.element_type)
 
     def flatmap(self, f):
+        """Map each element of the collection to a new collection, and flatten the results.
+
+        Parameters
+        ----------
+        f : callable
+            Function from element type of the collection to the type of the collection. For instance,
```
Comment (patrick-schultz) [inline python/hail/expr/expression.py]: Function from the element type ...

File: python/hail/expr/expression.py [hunk 3]
```diff
@@ -806,175 +1218,545 @@ def __radd__(self, other):
...
+        """Returns true if the polymorphism is a SNP.
+
+        Returns
+        -------
+        :class:`BooleanExpression`
+        """
         return self._method("isSNP", TBoolean())
 
     def is_star(self):
+        """Returns true if the polymorphism is not a star (upstream deletion) allele.
```
Comment (patrick-schultz) [inline python/hail/expr/expression.py]: Is that right? `is_star` returns true if not a star allele?

Comment (tpoterba) [reply python/hail/expr/expression.py]: will fix.

**Pull request merged**